src/number_theory/temporary_file.lean

```diff
@@ -0,0 +1,158 @@
+
+open polynomial localization alg_hom
+
+
+open_locale polynomial
+
+variables {R S : Type*} [comm_ring R] [comm_ring S]
+
+section move_me
+/- The results in this section are useful (they are direct generalisations of results in
+  `gauss_lemma.lean` but aren't necessary for this PR so will be shortly moved to another one) -/
+
+
+
+variables [algebra R S] {a : S} [is_domain S] [is_domain R] {φ : R →+* S} {f : R[X]}
+
+/- theorem is_primitive.is_unit_iff_is_unit_map_of_injective' (hinj : function.injective φ)
+  (hf : is_primitive f) : is_unit f ↔ is_unit (map φ f) :=
+begin
+  refine ⟨(map_ring_hom φ).is_unit_map, λ h, _⟩,
+  rcases is_unit_iff.1 h with ⟨_, ⟨u, rfl⟩, hu⟩,
+  have hdeg := degree_C u.ne_zero,
+  rw [hu, degree_map_eq_of_injective hinj] at hdeg,
+  rw [eq_C_of_degree_eq_zero hdeg] at hf,
+  rw [eq_C_of_degree_eq_zero hdeg, is_unit_C],
+  refine is_primitive_iff_is_unit_of_C_dvd.mp hf (f.coeff 0) (dvd_refl _),
+end
+
+lemma is_primitive_of_dvd' {p q : R[X]} (hp : is_primitive p) (hq : q ∣ p) : is_primitive q :=
+λ a ha, is_primitive_iff_is_unit_of_C_dvd.mp hp a (dvd_trans ha hq)
+
+lemma is_primitive.irreducible_of_irreducible_map_of_injective (hinj : function.injective φ)
+  (hf : is_primitive f) (h_irr : irreducible (map φ f)) :
+  irreducible f :=
+begin
+  refine ⟨λ h, h_irr.not_unit (is_unit.map (map_ring_hom φ) h), _⟩,
+  intros a b h,
+  rcases h_irr.is_unit_or_is_unit (by rw [h, polynomial.map_mul]) with hu | hu,
+  { left,
+    rwa is_primitive.is_unit_iff_is_unit_map_of_injective' hinj (is_primitive_of_dvd' hf
+      (dvd.intro _ h.symm)) },
+  right,
+  rwa is_primitive.is_unit_iff_is_unit_map_of_injective' hinj
+    (is_primitive_of_dvd' hf (dvd.intro_left _ h.symm))
+end
+
+end move_me -/
+
+local attribute [instance] frac_algebra_of_inj
+
+open euclidean_domain
+
+lemma polynomial.eq_zero_of_dvd_of_degree_lt {p q : R[X]} (h₁ : p ∣ q) (h₂ : degree q < degree p) :
+  q = 0 :=
+begin
+  by_contradiction hc,
+  exact (lt_iff_not_ge _ _ ).mp h₂ (degree_le_of_dvd h₁ hc),
+end
+
+lemma aeval_eq_zero_of_dvd_aeval_eq_zero {p q : R[X]} (h₁ : p ∣ q) {a : S} (h₂ : aeval a p = 0) :
+  aeval a q = 0 :=
+begin
+  rw [aeval_def, ← eval_map] at h₂ ⊢,
+  exact eval_eq_zero_of_dvd_of_eval_eq_zero (polynomial.map_dvd (algebra_map R S) h₁) h₂,
+end
+
+end preliminary_results
+
+local attribute [instance] frac_algebra_of_inj
+
+theorem algebra_map_minpoly_eq_minpoly [is_integrally_closed R]
+  [h : fact (function.injective (algebra_map R S))] (ha : is_integral R a) :
+  (map (algebra_map R (fraction_ring R)) (minpoly R a))
+    = minpoly (fraction_ring R) (algebra_map S (fraction_ring S) a) :=
+begin
+  --a few "trivial" preliminary results to set up the proof
+  have lem0 : minpoly (fraction_ring R) (algebra_map S (fraction_ring S) a) ∣
+    (map (algebra_map R (fraction_ring R)) (minpoly R a)),
+  { apply minpoly.dvd (fraction_ring R) (algebra_map S (fraction_ring S) a),
+    rw [← aeval_commuting_diagram, minpoly.aeval, map_zero],
```

Comment:
This theorem is where the `commuting_diagram` stuff are used, and indeed we have `[algebra R S]` here, so it seems the `commuting_diagram` stuff should be stated with `algebra` and `is_scalar_tower` rather than `ring_hom`s. Notice we have [fraction_ring.is_scalar_tower](https://leanprover-community.github.io/mathlib_docs/ring_theory/localization/fraction_ring.html#fraction_ring.is_scalar_tower) and you could get `no_smul_zero_divisors` from [function.injective.no_zero_smul_divisors](https://leanprover-community.github.io/mathlib_docs/algebra/module/basic.html#function.injective.no_zero_smul_divisors).

Reply:
Sure! I'll refactor my code to use `fraction_ring.is_scalar_tower`, and remove the extra "fraction algebra" definition I had added.